src/database/backends/mongo.ts

```diff
@@ -0,0 +1,85 @@
+import * as env from "env-var";
+import {
+  MongoClient,
+  MongoClientOptions,
+  InsertOneResult,
+  FindCursor,
+  Filter,
+  Document,
+} from "mongodb";
+
+const user = env.get("MONGO_USERNAME").required().asString();
+const pass = env.get("MONGO_PASSWORD").required().asString();
+const uri = env.get("MONGO_CLUSTER_URI").required().asString();
+const db = env.get("MONGO_DB").required().asString();
+
+const fullUri = `mongodb+srv://${user}:${pass}@${uri}/${db}`;
+
+const options: MongoClientOptions = {
+  connectTimeoutMS: 5000,
+  retryWrites: true,
+  w: "majority",
+};
+
+const client = new MongoClient(fullUri, options);
+
+// Will create the collection if it does not already exist
+export const insertOne = async <T>(
+  data: T,
+  db: string,
```

Comment:
`db` is a confusing name since it's not actually a database, but the "name of the database", perhaps rename this to `databaseName`